sparrow-py/pysrc/sparrow_py/_result.py

```diff
@@ -65,3 +66,32 @@ def iter_rows(self) -> Iterator[dict]:
             for row in next_batch.to_pylist():
                 yield row
             next_batch = self._ffi_execution.next_pyarrow()
+
+    async def iter_pandas_async(self) -> AsyncIterator[pd.DataFrame]:
```

Comment:
I think it may be possible to combine `iter_pandas` and `iter_pandas_async` by returning a class that implements both `__next__` and `__anext__`. Then if you try to use it asynchronously, it will call `__anext__` and you'll need to await it. If not, it will call `__next__` and it will block.

Pro: One method to document -- the thing that iterates over pandas.
Con: Seems like it could lead to really easy errors where you do `for ...` instead of `async for ...` and things end up blocking unexpectedly.

Looking around, I don't see cases for doing that, so I suspect we probably shouldn't unless we get some confirmation it makes sense / is a way people do things.

Reply:
Agreed -- would like to make it as hard as possible to make errors like that that are hard to diagnose, especially while we're still developing.